web/src/HomePage.js

```diff
@@ -99,52 +117,81 @@ class HomePage extends React.Component {
     return (
-      <Card size="small" title={
-        <div style={{width: "90vw"}}>
-          Edit Metadata&nbsp;&nbsp;&nbsp;&nbsp;
-          <Button type="primary" onClick={this.updateMetadata.bind(this)}>Save Change</Button>
-        </div>
-      } style={{marginLeft: '5px'}} type="inner">
-        <Row style={{marginTop: '20px'}} >
-          <Col style={{marginTop: '5px'}} span={2}>
+      <Card
+        size="small"
+        title={
+          <div style={{ width: "90vw" }}>
+            Edit Metadata&nbsp;&nbsp;&nbsp;&nbsp;
+            <Button type="primary" onClick={this.updateMetadata.bind(this)}>
+              Save Change
+            </Button>
+          </div>
+        }
+        style={{ marginLeft: "5px" }}
+        type="inner"
+      >
+        <Row style={{ marginTop: "20px" }}>
+          <Col style={{ marginTop: "5px" }} span={2}>
             Models:
           </Col>
-          <Col span={22} >
-            <ModelTable title="Models" table={this.state.models} onUpdateTable={this.onUpdateModels.bind(this)} />
+          <Col span={22}>
+            <ModelTable
+              title="Models"
+              table={this.state.models}
+              onUpdateTable={this.onUpdateModels.bind(this)}
+            />
           </Col>
         </Row>
-        <Row style={{marginTop: '20px'}} >
-          <Col style={{marginTop: '5px'}} span={2}>
+        <Row style={{ marginTop: "20px" }}>
+          <Col style={{ marginTop: "5px" }} span={2}>
             Adapters:
           </Col>
-          <Col span={22} >
-            <AdapterTable title="Adapters" table={this.state.adapters} onUpdateTable={this.onUpdateAdapters.bind(this)} />
+          <Col span={22}>
+            <AdapterTable
+              title="Adapters"
+              table={this.state.adapters}
+              onUpdateTable={this.onUpdateAdapters.bind(this)}
+            />
           </Col>
         </Row>
-        <Row style={{marginTop: '20px'}} >
-          <Col style={{marginTop: '5px'}} span={2}>
+        <Row style={{ marginTop: "20px" }}>
+          <Col style={{ marginTop: "5px" }} span={2}>
             Enforcers:
           </Col>
-          <Col span={22} >
-            <EnforcerTable title="Enforcers" table={this.state.enforcers} models={this.state.models === null ? [] : this.state.models} adapters={this.state.adapters === null ? [] : this.state.adapters} onUpdateTable={this.onUpdateEnforcers.bind(this)} />
+          <Col span={22}>
+            <EnforcerTable
+              title="Enforcers"
+              table={this.state.enforcers}
+              models={this.state.models === null ? [] : this.state.models}
+              adapters={this.state.adapters === null ? [] : this.state.adapters}
+              onUpdateTable={this.onUpdateEnforcers.bind(this)}
+            />
           </Col>
         </Row>
       </Card>
-    )
+    );
   }
 
   render() {
+    const { loading } = this.state;
+    if (loading) {
+      return (
+        <Loader

```

Comment:
I think the spinner is not on the right position it should be between the header and the footer

Reply:
I think it's in right position